### ISSUE TITLE
Array offset error in framework/View/Element/Html/Calendar.php

### ISSUE THREAD

**Issue by n2diving-dgx** [magento/magento2#33007]
Following an upgrade of M2.3.6-p1/php7.3  to M2.3.7/php7.4  an error similar to...

**main.CRITICAL: Notice: Trying to access array offset on value of type null in /vendor/magento/framework/View/Element/Html/Calendar.php on line 112** 

appears in the var/log/system.log each time the crontab runs, i.e. a few times per minute.  The cause is a  compatibility issue with Calendar.php that was identified and addressed in M2.4.0-beta... see the lengthy discussion in https://github.com/magento/magento2/issues/28729

Apparently the fix to Calendar.php was applied to M2.4.0 release, but the fix was not backported to M2.3.7 when the requirement for updating to php7.4 was mandated (which indirectly triggered an update to the offending version of the library).  Here is a .diff of essentially the M2.4.0 patch for the M2.3.7 version of Calendar, which resolved the problem in my M2.3.7 production environment.

```
--- a/View/Element/Html/Calendar.php
+++ b/View/Element/Html/Calendar.php
@@ -108,9 +108,7 @@
             ]
         );
 
-        // get "today" and "week" words
-        $this->assign('today', $this->encoder->encode($localeData['fields']['day']['relative']['0']));
-        $this->assign('week', $this->encoder->encode($localeData['fields']['week']['dn']));
+        $this->assignFieldsValues($localeData);
 
         // get "am" & "pm" words
         $this->assign('am', $this->encoder->encode($localeData['calendar']['gregorian']['AmPmMarkers']['0']));
@@ -190,4 +188,25 @@
         return (new \DateTime())->modify('- 100 years')->format('Y')
             . ':' . (new \DateTime())->modify('+ 100 years')->format('Y');
     }
+
+    /**
+     * Assign "fields" values from the ICU data
+     *
+     * @param \ResourceBundle $localeData
+     */
+    private function assignFieldsValues(\ResourceBundle $localeData): void
+    {
+        /**
+         * Fields value in the current position has been added to ICU Data tables
+         * starting with ICU library version 51.1.
+         * Due to fact that we do not use these variables in templates, we do not initialize them for older versions
+         *
+         * @see https://github.com/unicode-org/icu/blob/release-50-2/icu4c/source/data/locales/en.txt
+         * @see https://github.com/unicode-org/icu/blob/release-51-2/icu4c/source/data/locales/en.txt
+         */
+        if ($localeData->get('fields')) {
+            $this->assign('today', $this->encoder->encode($localeData['fields']['day']['relative']['0']));
+            $this->assign('week', $this->encoder->encode($localeData['fields']['week']['dn']));
+        }
+    }
 }

```

**Comment by mrtuvn** [comment]
cc: @gabrieldagama @sivaschenko  maybe this commit fix related issue not sync to 2.3.7

**Comment by ihor-sviziev** [comment]
@sivaschenko @sidolov @gabrieldagama, please add porting of https://github.com/magento/magento2/commit/c4ca054a7dbf1454a32bb256314958bac1ebd952 to the 2.3 release line.

**Comment by luccame** [comment]
Was this ever fixed in the 2.3 release line?

**Comment by engcom-Hotel** [comment]
Hello @n2diving-dgx,

We have moved this ticket to on hold for the PO confirmation.

Thanks
